src/config.rs

```diff
@@ -1,214 +1,35 @@
-		})
-		.collect()
-}
-
-impl LitteralTasks {
-	fn parse(self) -> TaskConf {
-		let handler = parse_cmd(&self.cmd);
-		let name = self.name.unwrap_or(handler[0].clone());
-		let args = to_bytes(handler);
-		let binary = args[0].clone();
-		let args = if args.len() == 0 { None } else { Some(args)};
-		let stdout = PathBuf::from(self.stdout.unwrap_or(format!("/tmp/{}.stdout", name)));
-		let stderr = PathBuf::from(self.stderr.unwrap_or(format!("/tmp/{}.stderr", name)));
-		TaskConf {
-			name,
-			binary,
-			args,
-			numproc: self.numproc,
-			umask: self.umask,
-			workingdir: self
-				.workingdir
-				.as_ref()
-				.and_then(|e| Some(PathBuf::from(e))),
-			autostart: self.autostart,
-			autorestart: self.autorestart.into(),
-			exitcodes: to_vec(self.exitcodes),
-			startretries: self.startretries,
-			startime: self.startime,
-			stopsignal: 9, // TODO: parse str into u32 or signal enum
-			stoptime: self.stoptime,
-			stdout,
-			stderr,
-			env: to_vec(self.env)
-				.into_iter()
-				.map(EnvVar::to_string)
-				.collect(),
-		}
-	}
-}
-
-#[derive(Deserialize, Debug)]
-struct LitteralConf {
-	port: Option<u32>,
-	tasks: Vec<LitteralTasks>,
-}
-
-impl LitteralConf {
-	fn parse(self) -> Conf {
-		Conf {
-			port: self.port.unwrap_or(6060),
-			tasks: self.tasks.into_iter().map(LitteralTasks::parse).collect(),
-		}
-	}
-}
+use crate::task::{ TaskConf, RunningTask };
+use crate::Error;
+use crate::parser;
 
 #[derive(Debug)]
 pub struct Conf {
-	port: u32,
-	tasks: Vec<TaskConf>,
+	pub port: u32,
+	pub tasks: Mutex<Vec<TaskConf>>, // Mutex or RWlock ?
+	pub runnings: Mutex<Vec<RunningTask>>
 }
 
 impl Conf {
 	pub fn new(path: String) -> Result<Conf, Error> {
-		let file = fs::read_to_string(path)?;
-		let litteral_conf = toml::from_str::<LitteralConf>(&file)?;
-		let conf = litteral_conf.parse();
-		Ok(conf)
+		parser::parse_config(path)
 	}
 
 	pub fn autostart(&self) {
-		for task in self.tasks.iter() {
+		let tasks = self.tasks.lock().unwrap();
+		let mut runnings = self.runnings.lock().unwrap();
+		for task in tasks.iter() {
 			if task.autostart == true {
-				task.run();
+				let pid = task.run(); // TODO: handle errors
+				runnings.push(RunningTask::new(&task.name, &task.name, pid));
 			}
 		}
 	}
-}
 
-#[cfg(test)]
-mod config_tests {
-	use super::*;
-
-	#[test]
-	fn parse_cmd_test() {
-		let tests: Vec<(&str, Vec<&str>)> = vec![
-			(r#"/bin/ls 'lol'"#, vec!["/bin/ls", "lol"]),
-			(r#"/bin/ls\ mdr"#, vec!["/bin/ls mdr"]),
-			(r#"/bin/ls '"lol'"#, vec!["/bin/ls", "\"lol"]),
-			(r#""'"\'"""#, vec![r#"''"#]),
-			// TODO: more tests
-		];
-		for test in tests {
-			assert_eq!(
-				parse_cmd(&test.0.to_string()),
-				test.1
-					.iter()
-					.map(|e| e.to_string())
-					.collect::<Vec<String>>()
-			)
-		}
+	pub fn dead_task(&self, pid: i32) {
+		let _tasks = self.tasks.lock().unwrap();
+		let mut _runnings = self.runnings.lock().unwrap();
+		println!("child died, pid: {}", pid);
```

Comment:
Je comprends bien comment est-ce qu'on arrive dans cette fonction depuis child_death dans signal.rs, mais si un enfant meurt, il ne faudrait pas le pop de _runnings?

Reply:
C'est l'issue #8, pour l'instant je ne rajoute ni enleve rien de running :)

Reply:
A si j'en rajoute attent pour moi, mais ce n'est pas le but de cette PR d'ajouter cette logique, juste de montrer comment on pourrait l'ajouté

Reply:
ca marche, je comprend bien, ca m'a juste sauté aux yeux, donc c'est noté!